File: code/JL_OTA_SDK/otasdk_java/src/main/java/com/jieli/otasdk_java/tool/IDeviceContract.java
package com.jieli.otasdk_java.tool;

import android.bluetooth.BluetoothDevice;

import com.jieli.otasdk_java.base.BasePresenter;
import com.jieli.otasdk_java.base.BaseView;

/**
 * @ClassName: IDeviceContract
 * @Description: java类作用描述
 * @Author: ZhangHuanMing
 * @CreateDate: 2021/12/23 9:55
 */
public interface IDeviceContract {
    public interface IDevScanPresenter extends BasePresenter {

        public boolean isScanning();

        public void startScan();

        public void stopScan();

        public BluetoothDevice getConnectedDevice();

        public void connectBtDevice(BluetoothDevice device);

        public void disconnectBtDevice(BluetoothDevice device);
    }

    public interface IDevScanView extends BaseView<IDevScanPresenter> {
        public void onScanStatus(int status, BluetoothDevice device);

        public void onConnectStatus(BluetoothDevice device, int status);

        public void onMandatoryUpgrade();

        public void onErrorCallback(int code, String message);

    }
}

File: code/JL_OTA_SDK/otasdk/src/main/java/com/jieli/otasdk/tool/IDeviceContract.kt
package com.jieli.otasdk.tool

import android.bluetooth.BluetoothDevice
import com.jieli.otasdk.base.BasePresenter
import com.jieli.otasdk.base.BaseView

/**
 * 设备操作接口
 *
 * @author zqjasonZhong
 * @date 2019/12/30
 */
interface IDeviceContract {

    interface IDevScanPresenter : BasePresenter {

        fun isScanning(): Boolean

        fun startScan()

        fun stopScan()

        fun getConnectedDevice(): BluetoothDevice?

        fun connectBtDevice(device: BluetoothDevice)

        fun disconnectBtDevice(device: BluetoothDevice?)
    }

    interface IDevScanView : BaseView<IDevScanPresenter> {
        fun onScanStatus(status: Int, device: BluetoothDevice?)

        fun onConnectStatus(device: BluetoothDevice?, status: Int)

        fun onMandatoryUpgrade()

        fun onErrorCallback(code: Int, message: String)

    }
}